rust/agents/relayer/src/msg/serial_submitter.rs

```diff
@@ -164,79 +171,118 @@ async fn receive_task(
 async fn prepare_task(
     domain: HyperlaneDomain,
     mut prepare_queue: OpQueue,
+    confirm_queue: OpQueue,
     tx_submit: mpsc::Sender<QueueOperation>,
     metrics: SerialSubmitterMetrics,
 ) {
+    // Pop as many ops as we can then send to the channel
+    let ops_to_prepare = tx_submit.capacity();
     loop {
-        // Pick the next message to try preparing.
-        let next = prepare_queue.pop().await;
-
-        let Some(Reverse(mut op)) = next else {
+        // Pop messages here according to the configured batch.
+        let mut batch = prepare_queue.pop_many(ops_to_prepare).await;
+        if batch.is_empty() {
             // queue is empty so give some time before checking again to prevent burning CPU
-            sleep(Duration::from_millis(200)).await;
+            sleep(Duration::from_millis(100)).await;
             continue;
-        };
-
-        trace!(?op, "Preparing operation");
-        debug_assert_eq!(*op.destination_domain(), domain);
-
-        match op.prepare().await {
-            PendingOperationResult::Success => {
-                debug!(?op, "Operation prepared");
-                metrics.ops_prepared.inc();
-                // this send will pause this task if the submitter is not ready to accept yet
-                if let Err(err) = tx_submit.send(op).await {
-                    tracing::error!(error=?err, "Failed to send prepared operation to submitter");
+        }
+        let mut task_prep_futures = vec![];
+        let op_refs = batch.iter_mut().map(|op| op.as_mut()).collect::<Vec<_>>();
+        for op in op_refs {
+            trace!(?op, "Preparing operation");
+            debug_assert_eq!(*op.destination_domain(), domain);
+            task_prep_futures.push(op.prepare());
+        }
+        let res = join_all(task_prep_futures).await;
+        let not_ready_count = res
+            .iter()
+            .filter(|r| {
+                matches!(
+                    r,
+                    PendingOperationResult::NotReady | PendingOperationResult::Reprepare
+                )
+            })
+            .count();
+        let batch_len = batch.len();
+        for (op, prepare_result) in batch.into_iter().zip(res.into_iter()) {
+            match prepare_result {
+                PendingOperationResult::Success => {
+                    debug!(?op, "Operation prepared");
+                    metrics.ops_prepared.inc();
+                    // this send will pause this task if the submitter is not ready to accept yet
+                    if let Err(err) = tx_submit.send(op).await {
+                        tracing::error!(error=?err, "Failed to send prepared operation to submitter");
```

Comment:
should be possible to adjust this to `tx_submit.send_many(batch.filter(success))`?